src/battle_util.c

```diff
@@ -2373,6 +2378,17 @@ u8 DoFieldEndTurnEffects(void)
                         BattleScriptExecute(BattleScript_TailwindEnds);
                         effect++;
                     }
+                    if (IsAbilityOnSide(gActiveBattler, ABILITY_WIND_RIDER))
```

Comment:
Is there a reason not to do it like this instead of adding a `various`? :eyes:

```diff
BattleScript_WindRiderLoopIter:
-	trywindrider BS_TARGET, BattleScript_WindRiderLoop_NextBattler	
+	jumpifability BS_TARGET, ABILITY_WIND_RIDER, BattleScript_WindRiderLoop_Activate
+	goto BattleScript_WindRiderLoop_Increment
+BattleScript_WindRiderLoop_Activate:
	call BattleScript_AbilityPopUp
	(...)
```

Reply:
yeah my bad it should have a `return` instead of `end`. A various/callnative is needed because wind power can activate if an ally uses tailwind too, so you need to compare sides, e.g.
```c
void BS_TryWindRider(void)
{
    u8 battler = GetBattlerForBattleScript(T1_READ_8(gBattlescriptCurrInstr + 5));
    if (GetBattlerSide(battler) == GetBattlerSide(gBattlerAttacker)
            && DoesBattleMonHaveAbility(battler, ABILITY_WIND_RIDER)) {
        RecordAbilityBattle(battler, ABILITY_WIND_RIDER);
        gBattleStruct->lastAbilities[battler] = ABILITY_WIND_RIDER;
        gBattlerAbility = gBattleScripting.battler = battler;
        gBattlescriptCurrInstr += 10;
    } else {
        gBattlescriptCurrInstr = T1_READ_PTR(gBattlescriptCurrInstr + 6); // fail ptr
    }
}
```

Reply:
> A various/callnative is needed because wind power can activate if an ally uses tailwind too, so you need to compare sides

I don't get it. Wouldn't `BattleScript_EffectTailwind` activate regardless of whether it's an ally or an opponent that uses Tailwind?

Reply:
yes. But opponent's wind power don't activate from tailwind so you need to check that the current loop battler is on the same side as the attacker

Reply:
Hm. Alright then. That does make sense. Any reason not to do it from within the Battle Script using the existing `getbattlerside` command though? :eyes:

```diff
BattleScript_WindRiderLoop_Iter:
+	getbattlerside BS_TARGET
+	jumpifbyte CMP_EQUAL, gBattleCommunication, B_SIDE_OPPONENT, BattleScript_WindRiderLoop_Increment
	jumpifability BS_TARGET, ABILITY_WIND_RIDER, BattleScript_WindRiderLoop_Activate
	goto BattleScript_WindRiderLoop_Increment
```